api/src/pcapi/alembic/versions/20240419T143429_154f5e68cb31_create_product_mediation.py

```diff
@@ -0,0 +1,51 @@
+"""
+create product_mediation table that will store the image url of the product
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+from pcapi.core.offers.models import TiteliveImageType
+from pcapi.utils.db import MagicEnum
+
+
+# pre/post deployment: pre
+# revision identifiers, used by Alembic.
+revision = "154f5e68cb31"
+down_revision = "d1767ee2dac1"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    op.execute("select 1 -- squawk:ignore-next-statement;")
+    op.create_table(
+        "product_mediation",
+        sa.Column("id", sa.BigInteger(), autoincrement=True, nullable=False),
+        sa.Column("idAtProviders", sa.String(length=70), nullable=True),
+        sa.Column("dateModifiedAtLastProvider", sa.DateTime(), nullable=True),
+        sa.Column("fieldsUpdated", postgresql.ARRAY(sa.String(length=100)), server_default="{}", nullable=False),
+        sa.Column("productId", sa.BigInteger(), nullable=False),
+        sa.Column("url", sa.String(length=255), nullable=False),
+        sa.Column("imageType", MagicEnum(TiteliveImageType), nullable=False),
+        sa.Column("lastProviderId", sa.BigInteger(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["lastProviderId"],
+            ["provider.id"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["productId"],
+            ["product.id"],
+        ),
```

Comment:
Ces 2 ajouts de FK prennent un lock `SHARE ROW EXCLUSIVE` sur les deux tables, on préfère séparer les migrations, surtout pour la FK sur la table `product`.

Pour éviter les timeouts lors du déploiement en production, on pourrait procéder ainsi:
1. ajout de table, sans les FK
2. ajout des FK not valid 
3. validation des FK

Reply:
je ne comprend pas pourquoi on demande un lock si la table est vide.
ic ma migration crée une table vide.